test/metabase/api/collection_test.clj

```diff
@@ -225,15 +225,86 @@
                                 response)))))))))
 
+(deftest collection-tree-shallow-test
+  (testing "GET /api/collection/tree?shallow=true"
+    (with-collection-hierarchy [a b c d e f g]
+      (let [personal-collection (collection/user->personal-collection (mt/user->id :rasta))
+            ids      (set (map :id (cons personal-collection [a b c d e f g])))]
+        (let [response (mt/user-http-request :rasta :get 200 "collection/tree?shallow=true")]
+          (testing "Make sure overall tree shape of the response is as is expected"
+            (is (= [{:name     "A"
+                     :children true}
+                    {:name "Rasta Toucan's Personal Collection"
+                     :children false}]
+                   (->> response
+                        (filter (fn [coll] (contains? ids (:id coll))))
+                        (map #(select-keys % [:name :children])))))
+            (testing "Make sure each Collection comes back with the expected keys"
+              (is (partial= {:description       nil
+                             :archived          false
+                             :entity_id         (:entity_id personal-collection)
+                             :slug              "rasta_toucan_s_personal_collection"
+                             :name              "Rasta Toucan's Personal Collection"
+                             :personal_owner_id (mt/user->id :rasta)
+                             :id                (u/the-id personal-collection)
+                             :location          "/"
+                             :namespace         nil
+                             :children          false
+                             :authority_level   nil}
+                            (some #(when (= (:id %) (u/the-id personal-collection)) %)
+                                  response))))))
+        (let [response (mt/user-http-request :rasta :get 200 (str "collection/tree?shallow=true&collection-id=" (:id a)))]
+          (testing "Make sure collection-id param works as expected"
+            (is (= [{:name     "B"
+                     :children false}
+                    {:name     "C"
+                     :children true}]
+                   (->> response
+                        (filter (fn [coll] (contains? ids (:id coll))))
+                        (map #(select-keys % [:name :children])))))))))))
+
+(deftest select-collections-shallow-test
+  (testing "Selecing collections based off collection-id equaling nil works."
+    (with-collection-hierarchy [a b c d e f g]
+      (let [personal-collection (collection/user->personal-collection (mt/user->id :crowberto))
+            ids      (set (map :id (cons personal-collection [a b c d e f g])))]
+        (mt/with-test-user :crowberto
+          (testing "Make sure we get the expected collections when collection-id is nil"
+            (let [collections (#'api.collection/select-collections nil nil nil true nil)]
```

Comment:
I think it would make more sense to pass the first two arguments as `false` since they're expected to be booleans.

Another option is to use `partial` here, since the first three args are constant for all the call sites in this test:
```
(let [select-collections (partial #'api.collection/select-collections false false nil)]
  ... 
  (let [collections (select-collections true nil)]
   ...))
```

That's a stylistic choice though, I don't have a strong opinion either way.